source/Main.cpp

```diff
@@ -40,11 +42,43 @@ int main(int argc, char* argv[]) {
   tool.add_options(options);
 
   try {
+    if (argc < 2) {
+      std::cerr << "Usage: " << argv[0] << " <json_config_file>\n";
+      return ExitCode::invalid_argument_error("No JSON configuration file provided.");
+    }
+    
+    std::string json_file_path = argv[1];
+
+    // Use JsonReader to parse the JSON file
+    Json::Value json = marianatrench::JsonReader::parse_json_file(json_file_path);
+    // Validate the JSON object
+    marianatrench::JsonValidation::validate_object(json);
+    // Use variables_map to store the JSON data
     program_options::variables_map variables;
-    program_options::store(
-        program_options::command_line_parser(argc, argv).options(options).run(),
-        variables);
+    // Populate variables_map from JSON data
+    for (const auto& key : json.getMemberNames()) {
```

Comment:
While this work, this is less than ideal. 

What we want instead is to have a static method `Options::from_json_file()`, without using the intermediate `program_options::variables_map`